### DISCUSSION TITLE
v4.3.0

### DISCUSSION THREAD

**Discussion by jsjoeio** [coder/code-server#5105] [Announcements]
v4.3.0

Code v1.65.2

### Changed

- Excluded .deb files from release Docker image which drops the compressed and
  uncompressed size by 58% and 34%.
- Upgraded to Code 1.65.2.

### Added

- Added a new CLI flag called `--disable-file-downloads` which allows you to
  disable the "Download..." option that shows in the UI when right-clicking on a
  file. This can also set by running `CS_DISABLE_FILE_DOWNLOADS=1`.
- Aligned the dependencies for binary and npm release artifacts.

### Fixed

- Fixed the code-server version from not displaying in the Help > About dialog.
- Fixed issues with the TypeScript and JavaScript Language Features Extension
  failing to activate.
- Fixed missing files in ipynb extension.
- Fixed the homebrew release workflow.
- Fixed the Docker release workflow from not always publishing version tags.

### New Contributors
* @azriel-stephen made their first contribution in https://github.com/coder/code-server/pull/4981
* @cjtim made their first contribution in https://github.com/coder/code-server/pull/5045
* @dwahler made their first contribution in https://github.com/coder/code-server/pull/5068
* @rileycalhoun made their first contribution in https://github.com/coder/code-server/pull/5079
* @neilnaveen made their first contribution in https://github.com/coder/code-server/pull/5090
* @thx2001r made their first contribution in https://github.com/coder/code-server/pull/5078

**Full Changelog**: https://github.com/coder/code-server/compare/v4.1.0...v4.3.0

<hr /><em>This discussion was created from the release <a href='https://github.com/coder/code-server/releases/tag/v4.3.0'>v4.3.0</a>.</em>

**Comment by notmrmario** [comment]
Hmm, the TypeScript and JavaScript Language Features Extension still doesn't work. I've tried installing older versions but still nothing. Downgrading code-server to v3.5.0 so ESLint doesnt activate seems to fix it to me. I haven't tested more recent code-server versions yet.

**Reply by jsjoeio** [reply]
Really? @code-asher and I tested it before we released it and it was working. Can you share any browser logs or anything that might be helpful? And also how you're using code-server (environment, platform, browser, etc.)

**Reply by notmrmario** [reply]
I will post the logs tomorrow, I cant at the moment, but I was getting the same error as in v4.2.0. As for the environment I use a Raspberry Pi with 64-bits Raspberry Pi OS as the host and I download it manually from the releases page, and for the browser I use Google Chrome in a Windows 10 computer.

**Reply by notmrmario** [reply]
I got this error after I restarted the ESLint server but I haven't been able to get it to appear again after restarting it again:
```
[Info  - 11:44:12] ESLint server is starting
[Info  - 11:44:13] ESLint server running in node v14.17.4
[Info  - 11:44:13] ESLint server is running.
[Info  - 11:44:14] 
Failed to load the ESLint library for the document /home/pi/Desktop/VS Code/JS/a.js

To use ESLint please install eslint by running npm install eslint in the workspace folder VS Code
or globally using 'npm install -g eslint'. You need to reopen the workspace after installing eslint.

If you are using yarn or pnpm instead of npm set the setting `eslint.packageManager` to either `yarn` or `pnpm`
Alternatively you can disable ESLint for the workspace folder VS Code by executing the 'Disable ESLint' command.
```

I also noticed an error in the browser console when I open a workspace file:
![image](https://user-images.githubusercontent.com/54030022/163670597-4d448862-38b1-4b30-ac8a-8fb201e86788.png)

I haven't got any further errors but while I was doing some research to try and fix the issue I found out some other users have got a double checkmark next to the ESLint status bar item while I don't.

**Reply by jsjoeio** [reply]
Hmm 🤔 I wonder if it's specific to Workspace files. If you remove the ESLint extension, do you still see this error in the browser?

**Reply by notmrmario** [reply]
Yup, it still shows the exact same error as before. I think you're right and it's specific to Workspace files.

**Reply by notmrmario** [reply]
Hmm, this is weird, I've just tested code-server v4.3.0 in a debian subsystem in termux and the extension seems to work just fine. My phone has the same architecture as the raspberry pi though. 🤔

**Reply by jsjoeio** [reply]
Interesting! I am not sure what it could be then 🤔

**Comment by drmrbrewer** [comment]
FYI I was seeing the typescript issue with the previous version (see [here](https://github.com/coder/code-server/issues/5019#issuecomment-1080272595)) and it's fixed for me in v4.3.0.

**Reply by notmrmario** [reply]
I think I've done everything correctly but it doesn't fix the issue for me. It shows the typescript package is installed but it doesn't do anything.

**Reply by jsjoeio** [reply]
Thank you for confirming @drmrbrewer!

**Comment by 502647092** [comment]
```
#5 [2/4] RUN npm install --unsafe-perm -g code-server
249#5 sha256:db2d5b02187499e02348ad4907c5b2aa573c8192c8335ab0f483406eacebee8d
250#5 25.10 npm ERR! code 1
251#5 25.10 npm ERR! path /usr/local/lib/node_modules/code-server/node_modules/argon2
252#5 25.10 npm ERR! command failed
253#5 25.10 npm ERR! command sh -c node-gyp rebuild
254#5 25.10 npm ERR! gyp info it worked if it ends with ok
255#5 25.10 npm ERR! gyp info using node-gyp@7.1.2
256#5 25.10 npm ERR! gyp info using node@14.19.0 | linux | x64
257#5 25.10 npm ERR! gyp info find Python using Python version 3.9.5 found at "/usr/bin/python3"
258#5 25.10 npm ERR! gyp http GET https://nodejs.org/download/release/v14.19.0/node-v14.19.0-headers.tar.gz
259#5 25.10 npm ERR! gyp http 200 https://nodejs.org/download/release/v14.19.0/node-v14.19.0-headers.tar.gz
260#5 25.10 npm ERR! gyp http GET https://nodejs.org/download/release/v14.19.0/SHASUMS256.txt
261#5 25.10 npm ERR! gyp http 200 https://nodejs.org/download/release/v14.19.0/SHASUMS256.txt
262#5 25.10 npm ERR! gyp info spawn /usr/bin/python3
263#5 25.10 npm ERR! gyp info spawn args [
264#5 25.10 npm ERR! gyp info spawn args '/usr/lib/node_modules/npm/node_modules/node-gyp/gyp/gyp_main.py',
265#5 25.10 npm ERR! gyp info spawn args 'binding.gyp',
266#5 25.10 npm ERR! gyp info spawn args '-f',
267#5 25.10 npm ERR! gyp info spawn args 'make',
268#5 25.10 npm ERR! gyp info spawn args '-I',
269#5 25.10 npm ERR! gyp info spawn args '/usr/local/lib/node_modules/code-server/node_modules/argon2/build/config.gypi',
270#5 25.10 npm ERR! gyp info spawn args '-I',
271#5 25.10 npm ERR! gyp info spawn args '/usr/lib/node_modules/npm/node_modules/node-gyp/addon.gypi',
272#5 25.10 npm ERR! gyp info spawn args '-I',
273#5 25.10 npm ERR! gyp info spawn args '/root/.cache/node-gyp/14.19.0/include/node/common.gypi',
274#5 25.10 npm ERR! gyp info spawn args '-Dlibrary=shared_library',
275#5 25.10 npm ERR! gyp info spawn args '-Dvisibility=default',
276#5 25.10 npm ERR! gyp info spawn args '-Dnode_root_dir=/root/.cache/node-gyp/14.19.0',
277#5 25.10 npm ERR! gyp info spawn args '-Dnode_gyp_dir=/usr/lib/node_modules/npm/node_modules/node-gyp',
278#5 25.10 npm ERR! gyp info spawn args '-Dnode_lib_file=/root/.cache/node-gyp/14.19.0/<(target_arch)/node.lib',
279#5 25.10 npm ERR! gyp info spawn args '-Dmodule_root_dir=/usr/local/lib/node_modules/code-server/node_modules/argon2',
280#5 25.10 npm ERR! gyp info spawn args '-Dnode_engine=v8',
281#5 25.10 npm ERR! gyp info spawn args '--depth=.',
282#5 25.10 npm ERR! gyp info spawn args '--no-parallel',
283#5 25.10 npm ERR! gyp info spawn args '--generator-output',
284#5 25.10 npm ERR! gyp info spawn args 'build',
285#5 25.10 npm ERR! gyp info spawn args '-Goutput_dir=.'
286#5 25.10 npm ERR! gyp info spawn args ]
287#5 25.10 npm ERR! internal/modules/cjs/loader.js:905
288#5 25.10 npm ERR! throw err;
289#5 25.10 npm ERR! ^
290#5 25.10 npm ERR!
291#5 25.10 npm ERR! Error: Cannot find module 'node-addon-api'
292#5 25.10 npm ERR! Require stack:
293#5 25.10 npm ERR! - /usr/local/lib/node_modules/code-server/node_modules/argon2/[eval]
294#5 25.10 npm ERR! at Function.Module._resolveFilename (internal/modules/cjs/loader.js:902:15)
295#5 25.10 npm ERR! at Function.Module._load (internal/modules/cjs/loader.js:746:27)
296#5 25.10 npm ERR! at Module.require (internal/modules/cjs/loader.js:974:19)
297#5 25.10 npm ERR! at require (internal/modules/cjs/helpers.js:101:18)
298#5 25.10 npm ERR! at [eval]:1:1
299#5 25.10 npm ERR! at Script.runInThisContext (vm.js:134:12)
300#5 25.10 npm ERR! at Object.runInThisContext (vm.js:310:38)
301#5 25.10 npm ERR! at internal/process/execution.js:81:19
302#5 25.10 npm ERR! at [eval]-wrapper:6:22
303#5 25.10 npm ERR! at evalScript (internal/process/execution.js:80:60) {
304#5 25.10 npm ERR! code: 'MODULE_NOT_FOUND',
305#5 25.10 npm ERR! requireStack: [
306#5 25.10 npm ERR! '/usr/local/lib/node_modules/code-server/node_modules/argon2/[eval]'
307#5 25.10 npm ERR! ]
308#5 25.10 npm ERR! }
309#5 25.10 npm ERR! gyp: Call to 'node -p "require('node-addon-api').include"' returned exit status 1 while in binding.gyp. while trying to load binding.gyp
310#5 25.10 npm ERR! gyp ERR! configure error
311#5 25.10 npm ERR! gyp ERR! stack Error: `gyp` failed with exit code: 1
312#5 25.10 npm ERR! gyp ERR! stack at ChildProcess.onCpExit (/usr/lib/node_modules/npm/node_modules/node-gyp/lib/configure.js:351:16)
313#5 25.10 npm ERR! gyp ERR! stack at ChildProcess.emit (events.js:400:28)
314#5 25.10 npm ERR! gyp ERR! stack at Process.ChildProcess._handle.onexit (internal/child_process.js:282:12)
315#5 25.10 npm ERR! gyp ERR! System Linux 4.19.24-7.29.cbp.al7.x86_64
316#5 25.10 npm ERR! gyp ERR! command "/usr/bin/node" "/usr/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
317#5 25.10 npm ERR! gyp ERR! cwd /usr/local/lib/node_modules/code-server/node_modules/argon2
318#5 25.10 npm ERR! gyp ERR! node -v v14.19.0
319#5 25.10 npm ERR! gyp ERR! node-gyp -v v7.1.2
320#5 25.10 npm ERR! gyp ERR! not ok
321#5 25.11
322#5 25.11 npm ERR! A complete log of this run can be found in:
323#5 25.11 npm ERR! /root/.npm/_logs/2022-04-18T15_18_16_193Z-debug.log
324#5 ERROR: executor failed running [/bin/sh -c npm install --unsafe-perm -g code-server]: buildkit-runc did not terminate successfully
```

**Reply by jsjoeio** [reply]
Thanks for the logs! What OS/architecture is this on?

**Reply by 502647092** [reply]
OS: alpine
Version: 3.14
arch: x86_64

**Reply by jsjoeio** [reply]
Did you get this figured out?
